modules/ml/src/main/java/org/apache/ignite/ml/util/generators/primitives/vector/VectorGeneratorPrimitives.java

```diff
@@ -0,0 +1,154 @@
+    }
+
+    /**
+     * Returns vector generator of 2D-vectors from ring-like distribution.
+     *
+     * @param radius Ring radius.
+     * @param fromAngle From angle.
+     * @param toAngle To angle.
+     * @return Generator.
+     */
+    public static VectorGenerator ring(double radius, double fromAngle, double toAngle) {
+        return ring(radius, fromAngle, toAngle, System.currentTimeMillis());
+    }
+
+    /**
+     * Returns vector generator of 2D-vectors from ring-like distribution around zero.
+     *
+     * @param radius Ring radius.
+     * @param fromAngle From angle.
+     * @param toAngle To angle.
+     * @param seed Seed.
+     * @return Generator.
+     */
+    public static VectorGenerator ring(double radius, double fromAngle, double toAngle, long seed) {
+        return new ParametricVectorGenerator(
+            new UniformRandomProducer(fromAngle, toAngle, seed),
+            t -> radius * Math.sin(t),
+            t -> radius * Math.cos(t)
+        );
+    }
+
+    /**
+     * Returns vector generator of vectors from multidimension uniform distribution around zero.
+     *
+     * @param bounds Parallelogram bounds.
+     * @return Generator.
+     */
+    public static VectorGenerator parallelogram(Vector bounds) {
+        return parallelogram(bounds, System.currentTimeMillis());
+    }
+
+    /**
+     * Returns vector generator of vectors from multidimension uniform distribution around zero.
+     *
+     * @param bounds Parallelogram bounds.
+     * @param seed Seed.
+     * @return Generator.
+     */
+    public static VectorGenerator parallelogram(Vector bounds, long seed) {
+        A.notEmpty(bounds.asArray(), "bounds.size() != 0");
+
+        UniformRandomProducer[] producers = new UniformRandomProducer[bounds.size()];
+        for (int i = 0; i < producers.length; i++)
+            producers[i] = new UniformRandomProducer(-bounds.get(i), bounds.get(i), seed >>= 2);
+
+        return RandomProducer.vectorize(producers);
+    }
+
+    /**
+     * Returns vector generator of 2D-vectors from circle-like distribution around zero.
+     *
+     * @param radius Circle radius.
+     * @return Generator.
+     */
+    public static VectorGenerator circle(double radius) {
+        return circle(radius, System.currentTimeMillis());
+    }
+
+    /**
+     * Returns vector generator of 2D-vectors from circle-like distribution around zero.
+     *
+     * @param radius Circle radius.
+     * @param seed Seed.
+     * @return Generator.
+     */
+    public static VectorGenerator circle(double radius, long seed) {
+        return new UniformRandomProducer(-radius, radius, seed)
+            .vectorize(2)
+            .filter(v -> Math.sqrt(v.getLengthSquared()) <= radius);
+    }
+
+    /**
+     * @param size Vector size.
+     * @return Generator of constant vector = zero.
+     */
+    public static VectorGenerator zero(int size) {
+        return constant(VectorUtils.of(new double[size]));
```

Comment:
I guess we could use `VectorUtils.zeroes(size)` here.

Reply:
thank you